src/App.tsx

```diff
@@ -1,223 +1,215 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React,
+{
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useCallback,
+} from 'react';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessages } from './components/ErrorMessages';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/Filter';
+import { TodoItem } from './components/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>('All');
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletingDataIds, setDeletingDataIds] = useState<number[]>([]);
+  const [updatingDataIds, setUpdatingDataIds] = useState<Todo[]>([]);
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setError('Can\'t load todos');
+        });
+    }
   }, []);
 
+  const remainingTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case 'Completed':
+        return todo.completed;
+
+      case 'Active':
+        return !todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const handleSubmitButton = (
+    e: React.FormEvent<HTMLFormElement>,
+  ) => {
+    e.preventDefault();
+
+    if (!title) {
+      setError('Title can\'t be empty');
```

Comment:
It is good practice to move error text to the enum
`setError('Title can\'t be empty');
setError(TodoErrors.EmptyTitle);`